src/client/Microsoft.Identity.Client/ApiConfig/AbstractConfidentialClientAcquireTokenParameterBuilder.cs

```diff
@@ -48,7 +48,8 @@ protected override void Validate()
             // Confidential client must have a credential
             if (ServiceBundle?.Config.ClientCredential == null &&
                 CommonParameters.OnBeforeTokenRequestHandler == null &&
-                ServiceBundle?.Config.AppTokenProvider == null
+                ServiceBundle?.Config.AppTokenProvider == null &&
+                ServiceBundle?.Config.IsManagedIdentity is false
```

Comment:
Since we are checking it in case Service Bundle is not null, the type of the variable becomes bool?. So using ! throws a compile time error: 
![image](https://user-images.githubusercontent.com/61847233/201729955-4fa6424f-045d-4b89-9f9a-1a3c72f9e1b9.png)